tokio-signal/src/unix.rs

```diff
@@ -84,6 +62,131 @@ impl Init for OsExtraData {
     }
 }
 
+/// Represents the specific kind of signal to listen for.
+#[derive(Debug, Clone, Copy)]
+pub struct SignalKind(c_int);
+
+impl SignalKind {
+    /// Allows for listening to any valid OS signal.
+    ///
+    /// For example, this can be used for listening for platform-specific
+    /// signals.
+    /// ```rust,no_run
+    /// # use tokio_signal::unix::SignalKind;
+    /// # let signum = -1;
+    /// // let signum = libc::OS_SPECIFIC_SIGNAL;
+    /// let kind = SignalKind::from_raw(signum);
+    /// ```
+    pub fn from_raw(signum: c_int) -> Self {
+        Self(signum)
+    }
+
+    /// Represents the SIGALRM signal.
+    ///
+    /// On Unix systems this signal is sent when a real-time timer has expired.
+    /// By default, the process is terminated by this signal.
+    pub fn sigalrm() -> Self {
```

Comment:
I would probably avoid prefacing these fns w/ `signal`. Since the type is named `SignalKind`, using a signal prefix would be a stutter. Maybe use the full name, like `fn alarm`()`.